metaflow/cli.py

```diff
@@ -901,6 +899,12 @@ def version(obj):
     type=click.Choice(MONITOR_SIDECARS),
     help="Monitoring backend type",
 )
+@click.option(
```

Comment:
I'm not a fan of having options cause other options to be ignored with no warning. I would add a check to see if there are conflicting options. `dry-run` is also a bit misleading (it's not dry run and things may happen, for example conda package caching, anything the code does as well, etc).